### ISSUE TITLE
FR: Add a per-schedule option to disable logging runs

### ISSUE THREAD

**Issue by rauanmayemir** [citusdata/pg_cron#366]
With a new feature allowing to schedule jobs with second granularity, `job_run_details` is clogged very-very fast rendering it practically useless, especially if you do want to keep log run details for your critical jobs and discard them for routine queries.

Adding an argument to disable logging per-schedule would be very helpful:

```
cron.schedule(job_name text, schedule text, command text, log boolean)
```

**Comment by TsinghuaLucky912** [comment]
Hello, I am quite interested in this. What I want to know is, do you think the reason for the blockage is because of writing logs?

Can you provide a case? Thank you

**Comment by rauanmayemir** [comment]
What blockage? My main concern is not about log inserts adding _some_ kind of load on the server (although it's probably a valid point, too), but purely convenience. 1 job run per second will instantly render `job_run_details` not very useful as most of the runs will be those second-granilarity schedules.

**Comment by Splarv** [comment]
Yep, I vote for this. The reason in same. We have important cron job launched rare and want to see log record about this. And several 1 second jobs to emulate one time jobs. They simply read one time jobs from the queue table and execute it. There will be huge amound of useless record. Will be convenient to to disable this per job.

**Comment by sfc-gh-mslot** [comment]
Yes, seems like a good idea. Anyone willing to throw Claude at this? :)

**Comment by cpruijsen** [comment]
I have an implementation of this and would like to confirm the intended shape before opening a PR, since the two readings in this thread differ in behaviour rather than just in naming.

1. A boolean per-job flag mirroring the existing `cron.log_run` GUC, so `cron.schedule(..., log_run := false)` suppresses all `cron.job_run_details` writes for that job.
2. The enum suggested later in the thread (`all` / `errors` / `off`), so failures stay recorded.

The enum needs a decision this thread does not settle. A row is written when a job starts and then updated as it progresses, across the six `CronLogRun` sites in `src/pg_cron.c`. An errors-only mode is therefore either insert-then-delete-on-success, or defer the insert until the outcome is known and give up visibility of in-flight jobs. Both are reasonable, and they are different features, so I would rather not guess.

One consequence that applies either way: adding a parameter to `cron.alter_job` means dropping and recreating it, so any `GRANT EXECUTE` on the current `cron.alter_job(bigint,text,text,text,text,boolean)` will not carry over to the new signature. I can note that in the upgrade script, but it is worth flagging in case it affects which shape you prefer.

Happy to build whichever you would rather have.

**Comment by cpruijsen** [comment]
Opened #442 with the boolean version, since the thread had gone quiet and a patch is easier to react to than a question. It is built and tested against PostgreSQL 17.11, including a live launcher check that `log_run := false` writes no `job_run_details` rows.

If you would rather have the `all` / `errors` / `off` enum, say so and I will rework it. The open question there is still whether errors-only deletes the row on success or defers the insert and loses in-flight visibility.

**Comment by rauanmayemir** [comment]
@cpruijsen who are you addressing 'you' to?

I think the last comment by @sfc-gh-mslot blessed the enum. ;)
